### PR TITLE
Modify sklearn example

### PR DESCRIPTION
**Pull request by nikp1172**

### CODE REVIEW

**Review by chiragjn — APPROVED** [review]

File: iris-flower-classification-sklearn/service/main.py [hunk 1]
```diff
@@ -22,10 +23,11 @@ def _get_model():
...
     if _model is None:
         api_key = os.environ.get('TFY_API_KEY')
-        run_id = os.environ.get('TFY_RUN_ID')
-        client = mlf.get_client(api_key=api_key)
-        run = client.get_run(run_id)
-        _model = run.get_model()
+        host=os.environ.get('TFY_HOST')
+        model_version_fqn = os.environ.get('TFY_MODEL_VERSION_FQN')
+        client = mlf.get_client(tracking_uri=host, api_key=api_key)
+        _model = client.get_model(model_version_fqn).load()
```
Comment (chiragjn) [inline iris-flower-classification-sklearn/service/main.py:29]: now we don't need to pass these explicitly if we follow `TFY_API_KEY` and `TFY_HOST` names
so just
```suggestion
        client = mlf.get_client()
        _model = client.get_model(model_version_fqn).load()
```

File: iris-flower-classification-sklearn/service/requirements.txt [hunk 2]
```diff
@@ -1,10 +1,5 @@
...
-numpy==1.21.6
 pandas==1.1.5
 scikit-learn==0.24.2
-shap==0.40.0 
 fastapi==0.78.0
-prometheus-client==0.14.1
 uvicorn==0.16.0
-mlfoundry>=0.3.33,<0.4.0
-servicefoundry>=0.1.69,<0.2.0
+mlfoundry>=0.4.5,<0.5
```
Comment (chiragjn) [inline iris-flower-classification-sklearn/service/requirements.txt:5]: not necessary

```suggestion
mlfoundry>=0.4.6,<0.5
```

File: iris-flower-classification-sklearn/service/servicefoundry.yaml [hunk 3]
```diff
@@ -8,16 +8,13 @@ components:
...
         type: tfy-python-buildpack
-        python_version: 3.9
         command: uvicorn main:app --port 8000 --host 0.0.0.0
-    resources:
-      cpu_request: 50m
-      cpu_limit: 100m
-      memory_request: 128Mi
-      memorty_limit: 512Mi
+    env:
+      TFY_MODEL_VERSION_FQN: model:truefoundry/user-truefoundry/iris-clf-demo/iris-model-demo:1
```
Comment (chiragjn) [inline iris-flower-classification-sklearn/service/servicefoundry.yaml:13]: Is this project public to fetch this model or do we need to instruct to paste fqn here?

File: iris-flower-classification-sklearn/train.ipynb [hunk 4]
```diff
@@ -69,7 +69,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "client = mlf.get_client(api_key=MLF_API_KEY)"
+    "client = mlf.get_client(api_key=api_key)"
```
Comment (chiragjn) [inline iris-flower-classification-sklearn/train.ipynb:72]: do we need to pass host here?

Comment (chiragjn) [reply iris-flower-classification-sklearn/train.ipynb:72]: actually let's update this cell to just 
```
client = mlf.get_client()
```
and let the user login interactively

we can delete the cell above this

Comment (chiragjn) [reply iris-flower-classification-sklearn/train.ipynb:72]: In the `! pip install -U mlfoundry` cell add constraints

```
! pip install -U "mlfoundry>=0.4.6,<0.5.0"
```

File: iris-flower-classification-sklearn/service/servicefoundry.yaml [hunk 5]
```diff
@@ -8,16 +8,13 @@ components:
...
+      TFY_API_KEY: <paste your api key here>
+      TFY_HOST: <paste the tfy host, eg: https://app.truefoundry.com>
```
Comment (chiragjn) [inline iris-flower-classification-sklearn/service/servicefoundry.yaml:15]: we can remove this for now to keep things minimal

File: iris-flower-classification-sklearn/train.ipynb [hunk 6]
```diff
@@ -238,7 +238,12 @@
     "print(clf.get_params())\n",
     "run.log_params(clf.get_params())\n",
     "run.log_params({'classes': clf.classes_, 'features': clf.n_features_in_})\n",
-    "run.log_model(clf, framework=mlf.ModelFramework.SKLEARN)"
+    "run.log_model(\n",
+    "    name=\"iris-model-demo\",\n",
+    "    model=clf,\n",
+    "    framework=mlf.ModelFramework.SKLEARN,\n",
+    "    description=\"sklearn framework used\"\n",
+    ")"
```
Comment (chiragjn) [inline iris-flower-classification-sklearn/train.ipynb:246]: add printing model version fqn here
```
model_version = run.log_model(...
)
print(model_version.fqn)
```

File: iris-flower-classification-sklearn/service/servicefoundry.yaml (hunk 5 above)
Comment (nikp1172) [reply iris-flower-classification-sklearn/service/servicefoundry.yaml:15]: keeping it bcoz currently people are using develop
it should be simple for them to add......

**Pull request merged**